pkg/scc/scc.go

```diff
@@ -184,24 +268,94 @@ func (builder *Builder) WithAllowCapabilities(allowCapabilities []coreV1.Capabil
 	return builder
 }
 
+// WithDefaultAddCapabilities adds list of defaultAddCapabilities to SecurityContextConstraints.
+func (builder *Builder) WithDefaultAddCapabilities(defaultAddCapabilities []coreV1.Capability) *Builder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Redefining SecurityContextConstraints %s with"+
+		"defaultAddCapabilities: %v", builder.Definition.Name, defaultAddCapabilities)
+
+	if builder.Definition.DefaultAddCapabilities == nil {
+		builder.Definition.DefaultAddCapabilities = defaultAddCapabilities
+
+		return builder
+	}
+
+	builder.Definition.DefaultAddCapabilities = append(builder.Definition.DefaultAddCapabilities,
+		defaultAddCapabilities...)
+
+	return builder
+}
+
+// WithPriority adds priority to SecurityContextConstraints.
+func (builder *Builder) WithPriority(priority *int32) *Builder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Redefining SecurityContextConstraints %s with"+
+		"priority: %v", builder.Definition.Name, priority)
+
+	builder.Definition.Priority = priority
+
+	return builder
+}
+
 // WithFSGroup adds fsGroup to SecurityContextConstraints.
-func (builder *Builder) WithFSGroup(fsGroup string) *Builder {
+func (builder *Builder) WithFSGroup(fsGroupType string, fsGroupRange map[string]int64) *Builder {
 	if valid, _ := builder.validate(); !valid {
 		return builder
 	}
 
 	glog.V(100).Infof("Redefining SecurityContextConstraints %s with"+
-		"fsGroup: %s", builder.Definition.Name, fsGroup)
+		"fsGroupType: %s", builder.Definition.Name, fsGroupType)
 
-	if fsGroup == "" {
-		glog.V(100).Infof("SecurityContextConstraints 'fsGroup' argument cannot be empty")
+	if fsGroupType == "" {
+		glog.V(100).Infof("SecurityContextConstraints 'fsGroupType' argument cannot be empty")
 
-		builder.errorMsg = "SecurityContextConstraints 'fsGroup' cannot be empty string"
+		builder.errorMsg = "SecurityContextConstraints 'fsGroupType' cannot be empty string"
 
 		return builder
 	}
 
-	builder.Definition.FSGroup.Type = securityV1.FSGroupStrategyType(fsGroup)
+	builder.Definition.FSGroup.Type = securityV1.FSGroupStrategyType(fsGroupType)
+
+	if fsGroupRange == nil {
```

Comment:
In golang we can't verify is map is empty by this condition. Please use if len(fsGroupRange) == 0 instead.